client.go

```diff
@@ -352,7 +352,7 @@ func (client *Client) Recover(err interface{}, hint *EventHint, scope EventModif
 	// is store the Context in the EventHint and there nil means the Context is
 	// not available.
 	//nolint: staticcheck
-	return client.RecoverWithContext(nil, err, hint, scope)
+	return client.RecoverWithContext(context.TODO(), err, hint, scope)
```

Comment:
Changing this would require accommodating changes in other places. In `RecoverWithContext` we do account for context being nil, and the context ends up in EventHint.Context.

Reply:
I will revert this change if you want but I'd also like you to have a look at [this](https://deepsource.io/gh/withshubh/sentry-go/issue/SCC-SA1012/description/).

Reply:
I understand normal code [should not pass nil `Context` values](https://golang.org/pkg/context/), and that's what automatic code analysis tools will look for, but that's not how the code here was originally implemented.

As I pointed earlier, a change here would require changes in other places. `RecoverWithContext` itself handles `nil` and does not use the `Context` for cancellations nor any of the normal use cases, instead all it does is populate the `EventHint.Context` field (which might be `nil` when we don't know the `Context`).

We could change `EventHint.Context` to give a guarantee it is never `nil`, but that has not been anyone's ask (and that's why I wanted to know if you were trying to solve any particular problem with this PR).

I'll revert this and merge the other two changes.

```suggestion
	return client.RecoverWithContext(nil, err, hint, scope)
```